### DISCUSSION TITLE
Can not get the currently connected chain when connecting Safe through WalletConnect

### DISCUSSION THREAD

**Discussion by einaralex** [rainbow-me/rainbowkit#1912] [Q&A]
Here is an example of our setup: [CODESANDBOX](https://codesandbox.io/p/sandbox/rainbowkit-sign-3sth7y?file=%2Fsrc%2FApp.js%3A16%2C40&layout=%257B%2522sidebarPanel%2522%253A%2522EXPLORER%2522%252C%2522rootPanelGroup%2522%253A%257B%2522direction%2522%253A%2522horizontal%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522id%2522%253A%2522ROOT_LAYOUT%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522clu95920d00063b6u9rnc41nk%2522%252C%2522sizes%2522%253A%255B100%252C0%255D%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522EDITOR%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522id%2522%253A%2522clu95920c00023b6uly5uw3ch%2522%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522SHELLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522id%2522%253A%2522clu95920c00033b6uay6hvuko%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522DEVTOOLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522id%2522%253A%2522clu95920c00053b6uelufjjuh%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%252C%2522sizes%2522%253A%255B40%252C60%255D%257D%252C%2522tabbedPanels%2522%253A%257B%2522clu95920c00023b6uly5uw3ch%2522%253A%257B%2522id%2522%253A%2522clu95920c00023b6uly5uw3ch%2522%252C%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clu96g0le00023b6um53w7235%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522FILE%2522%252C%2522filepath%2522%253A%2522%252Fsrc%252FWalletProvider.js%2522%252C%2522state%2522%253A%2522IDLE%2522%252C%2522initialSelections%2522%253A%255B%257B%2522startLineNumber%2522%253A48%252C%2522startColumn%2522%253A31%252C%2522endLineNumber%2522%253A48%252C%2522endColumn%2522%253A31%257D%255D%257D%252C%257B%2522id%2522%253A%2522clu97ich600023b6u8o9614t9%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522FILE%2522%252C%2522initialSelections%2522%253A%255B%257B%2522startLineNumber%2522%253A12%252C%2522startColumn%2522%253A22%252C%2522endLineNumber%2522%253A12%252C%2522endColumn%2522%253A22%257D%255D%252C%2522filepath%2522%253A%2522%252Fpackage.json%2522%252C%2522state%2522%253A%2522IDLE%2522%257D%252C%257B%2522id%2522%253A%2522cluk1v64b00023b6mkayxnjtr%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522FILE%2522%252C%2522initialSelections%2522%253A%255B%257B%2522startLineNumber%2522%253A16%252C%2522startColumn%2522%253A40%252C%2522endLineNumber%2522%253A16%252C%2522endColumn%2522%253A40%257D%255D%252C%2522filepath%2522%253A%2522%252Fsrc%252FApp.js%2522%252C%2522state%2522%253A%2522IDLE%2522%257D%255D%252C%2522activeTabId%2522%253A%2522cluk1v64b00023b6mkayxnjtr%2522%257D%252C%2522clu95920c00053b6uelufjjuh%2522%253A%257B%2522id%2522%253A%2522clu95920c00053b6uelufjjuh%2522%252C%2522activeTabId%2522%253A%2522clu95m1br006e3b6uo7dmc0ww%2522%252C%2522tabs%2522%253A%255B%257B%2522type%2522%253A%2522UNASSIGNED_PORT%2522%252C%2522port%2522%253A0%252C%2522id%2522%253A%2522clu95m1br006e3b6uo7dmc0ww%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522path%2522%253A%2522%252F%2522%257D%252C%257B%2522type%2522%253A%2522ASSISTANT_CHAT%2522%252C%2522chatId%2522%253A%2522clu99dfiq011s3b6ut5xv6cxa%2522%252C%2522id%2522%253A%2522clu99dfis011u3b6ucer6l2jc%2522%252C%2522mode%2522%253A%2522permanent%2522%257D%255D%257D%252C%2522clu95920c00033b6uay6hvuko%2522%253A%257B%2522tabs%2522%253A%255B%255D%252C%2522id%2522%253A%2522clu95920c00033b6uay6hvuko%2522%257D%257D%252C%2522showDevtools%2522%253Atrue%252C%2522showShells%2522%253Afalse%252C%2522showSidebar%2522%253Atrue%252C%2522sidebarPanelSize%2522%253A15.403551251008892%257D)

I try connecting with a Sepolia wallet, but it will always default to Gnosis Chiado because it's the first one in the chain list.
As I understood, useNetwork should show the wallets chain, but this example shows that it's not working correctly.

1. Create Sepolia Safe on https://app.safe.global/
2. Go to Home  > Use WalletConnect and pair the Safe to the Codesandbox.
3. Observer that chain from useNetwork is Gnosis Chiado, not Sepolia.

**Comment by 0xmkh** [comment]
Safe has a few issues since it's using smart account addresses + whatever chain safe wants to connect to first through WalletConnect it's the same issue that's something we cannot control. All the other wallets works totally fine apart from MetaMask sometimes, but feel free to raise an issue in the wagmi repo if you think this is a real issue.

Wagmi's repo: https://github.com/wevm/wagmi
